doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,785 @@
+ssl_key_file  = '/etc/xroad/postgresql/server.key'
+
+listen_addresses = *    # (default is localhost. Alternatively: localhost, <IP of the interface the slaves connect to>")
+wal_level = hot_standby
+max_wal_senders   = 3   # should be ~ number of slaves plus some small number. Here, we assume there are two slaves.
+wal_keep_segments = 8   # keep some wal segments so that slaves that are offline can catch up.
+```
+
+For more information about the streaming replication configuration options,
+see the [PostgreSQL documentation](https://www.postgresql.org/docs/9.2/static/runtime-config-replication.html).
+
+
+Edit `pg_hba.conf` and enable connections to the replication pseudo database using client certificates. See chapter
+[4.1](#41-setting-up-tls-certificates-for-database-authentication) for the authentication setup.
+
+```
+hostssl     replication     +slavenode  samenet     cert
+```
+**Note:** The CN field in the certificate subject must match a replication user name in postgresql. See the [PostgreSQL
+ documentation](https://www.postgresql.org/docs/9.3/static/auth-pg-hba-conf.html) for more details.
+
+The `samenet` above assumes that the slaves will be in the same subnet as the master.
+
+Start the master instance:
+
+**Ubuntu:**
+
+```bash
+/etc/init.d/postgresql start
+```
+
+**RHEL:**
+
+```bash
+systemctl start postgresql-serverconf
+```
+
+Create the replication user(s) with password authentication disabled:
+```bash
+sudo -u postgres psql -p 5433 -c "CREATE ROLE slavenode NOLOGIN";
+sudo -u postgres psql -p 5433 -c "CREATE USER "<nodename>" REPLICATION PASSWORD NULL IN ROLE slavenode";
+```
+
+Create a user named `serverconf` for local `serverconf` database access:
+
+```bash
+sudo -u postgres psql -p 5433 -c "CREATE USER serverconf PASSWORD '<password>'";
+```
+
+Copy the `serverconf` database from the default instance to the new instance:
+
+```bash
+sudo -u postgres pg_dump -C serverconf | sudo -u postgres psql -p 5433 -f -
+```
+
+To avoid confusion, the *old* `serverconf` database on the master should be renamed, or even deleted.
+```bash
+sudo -u postgres psql -p 5432 -c "ALTER DATABASE serverconf RENAME TO serverconf_old";
+```
+
+
+### 4.5 Configuring the slave instance for replication
+
+Prerequisites:
+* A separate postgresql instance has been created.
+* TLS keys and certificates have been configured in `/etc/xroad/postgresql` as described in section
+[4.1 Setting up TLS certificates for database authentication](#41-setting-up-tls-certificates-for-database-authentication)
+
+
+Go to the postgresql data directory:
+ * RHEL: `/var/lib/pgsql/serverconf`
+ * Ubuntu: `/var/lib/postgresql/9.3/serverconf`
+
+Clear the data directory:
+
+ ```bash
+ rm -rf *
+ ```
+
+Do a base backup with `pg_basebackup`:
+```bash
+sudo -u postgres PGSSLMODE=verify-ca PGSSLROOTCERT=/etc/xroad/postgresql/ca.crt PGSSLCERT=/etc/xroad/postgresql/server.crt PGSSLKEY=/etc/xroad/postgresql/server.key pg_basebackup -h <master> -p 5433 -U <nodename> -D .
```

Comment:
maybe remind that master / nodename values should be filled in (also everywhere where similar replacement should be done)